m_general.R

```diff
@@ -286,43 +289,71 @@ data_for_analysis_server <- function(id,
 
 # Dataset import  ------------------------ ---------------------------------
 dataset_import_server <- function(id,
-                                  QA, Cov, QA_Cov, Standard) {
-  
+                                  defaultData) {
+
   moduleServer(
     id,
     function(input, output, session) {
-      
-      # Make the data file that was uploaded reactive 
-      output$data <- reactive({ 
-        
-                  file1 <- input$data_input
-                  
-                  if (is.null(file1)) {
-                    if ('2' %in% input$default) {
-                      return(QA %>% dplyr::mutate(year.cts = year, prevalence.cts = round((TP+FN)/(TP+FN+FP+TN), 3)))
-                    }
-                    if ('3' %in% input$default) {
-                      return(Cov %>% dplyr::mutate(year.cts = year, prevalence.cts = round((TP+FN)/(TP+FN+FP+TN), 3)))
-                    }
-                    if ('4' %in% input$default) {
-                      return(QA_Cov %>% dplyr::mutate(year.cts = year, prevalence.cts = round((TP+FN)/(TP+FN+FP+TN), 3)))
-                    }
-                    else {
-                      return(Standard %>% dplyr::mutate(year.cts = year, prevalence.cts = round((TP+FN)/(TP+FN+FP+TN), 3)) )
-                    }
-                  }
-                  
-                  else { 
-                    a <- read.table(file = file1$datapath, 
-                                    sep = input$sep,
-                                    header = input$header, 
-                                    stringsAsFactors = FALSE) %>% 
-                      dplyr::mutate(year.cts = year, prevalence.cts = round((TP+FN)/(TP+FN+FP+TN), 3))
-                    a
-                  }
-        
+
+      # Create a definable reactive value to allow reloading of data
+      reload <- reactiveVal(FALSE)
+
+      # Initalise input field
+      output$data_input <- renderUI(DefaultFileInput(id))
```

Comment:
Should this be using the same ID as the parent module?

Reply:
No, but this is based on previous code, and will need to be refactored when all the modules get their own IDs.